server/src/main/java/io/envoyproxy/controlplane/server/LatestDiscoveryResponse.java

```diff
@@ -0,0 +1,49 @@
+package io.envoyproxy.controlplane.server;
+
+import java.util.Objects;
+import java.util.Set;
+
+class LatestDiscoveryResponse {
+
+  private final String nonce;
+  private final Set<String> resources;
```

Comment:
nitpick: `resourceNames`?
also - maybe leave a comment on why this class exists, meaning this is optimization.

Reply:
Added comment and changed name